### ISSUE TITLE
Markdown gives `[remark-lint-no-shortcut-reference-link]` and `[remark-lint-no-undefined-references]` false positives

### ISSUE THREAD

**Issue by SwuduSusuwu** [remarkjs/remark-lint#329]
### Initial checklist

- [x] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [x] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [x] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [x] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=issues) and [discussions](https://github.com/orgs/remarkjs/discussions) and couldn’t find anything (or linked relevant results below)

### Affected package

`remark-lint` (whichever version _GitHub_'s [_Codacy_](https://github.com/marketplace/codacy) workflow uses). 

### Steps to reproduce

https://github.com/SwuduSusuwu/SubStack/pull/46/commits/ce29f991b8ac3978ae59f2bfccc6ebfa5013d427#diff-f6ed156e4bf5c791680662464b94ea5d753f219ee816b385f67870e2c0d7d4c7

### Actual behavior

https://github.com/SwuduSusuwu/SubStack/security/code-scanning/2664

### Expected behavior

Since the form from https://github.github.com/gfm/#links is used,
> All checks have passed

### Runtime

_GitHub_ workflow

### Package manager

_GitHub_ workflow

### Operating system

_GitHub_ workflow

**Comment by SwuduSusuwu** [comment]
Also:
- https://github.com/SwuduSusuwu/SubStack/security/code-scanning/2660
- https://github.com/SwuduSusuwu/SubStack/security/code-scanning/2661
- https://github.com/SwuduSusuwu/SubStack/security/code-scanning/2662
- https://github.com/SwuduSusuwu/SubStack/security/code-scanning/2663

**Comment by JounQin** [comment]
@SwuduSusuwu Your provided links are all private.

**Comment by SwuduSusuwu** [comment]
> [@SwuduSusuwu](https://github.com/SwuduSusuwu) Your provided links are all private.

Oops, did not know. Thought that if the repository was public, that the results of open source scan tools were public (since anyone can use those on the public repository's source code).
https://duckduckgo.com/?q=How+to+set+GitHub+%22code+scanning+alerts%22+to+public has;
- https://docs.github.com/en/code-security/code-scanning/managing-code-scanning-alerts/about-code-scanning-alerts (which does not have a section on how to set those to public)
- [duck.ai_2025-04-10_08-52-52.txt](https://github.com/user-attachments/files/19687923/duck.ai_2025-04-10_08-52-52.txt) (which says that those are supposed to default to public)
- Numerous irrelevent results.

How to have those public?

**Comment by SwuduSusuwu** [comment]
> Hi! Thanks for taking the time to contribute! This has been marked by a maintainer as needing a reproduction: It’s not yet clear whether this is a problem. Here are a couple tips:
> 
>     * Thoroughly document how to reproduce the problem, in steps or with code
> 
>     * Don’t link to your complete project: make the repro as tiny as possible, preferrably with only the problematic project in question

Upon further review, this is due to normal [Pandoc footnotes](https://pandoc.org/chunkedhtml-demo/8.19-footnotes.html) (which is all that GitHub allows, but which `remark-lint` for whatever reason [chooses not to allow](https://github.com/remarkjs/remark-lint/issues/234).)
Since `remark-lint` is ran from _Codacy_'s workflow (https://github.com/codacy/codacy-analysis-cli-action), it is not possible to alter those defaults (such as to [enable the module which allows Pandoc footnotes](https://github.com/remarkjs/remark-lint/issues/234#issuecomment-629099478)).

Had dismissed those before, and the new commit (which just adds a normal link to, modified the line with those _Pandoc_ footnotes, which re-triggered the false positive and led me to conclude that normal links were being rejected by `remark-lint`).

Still wish that `remark-lint` allows _Pandoc_ footnotes (which are the most used).

**Comment by SwuduSusuwu** [comment]
> [@SwuduSusuwu](https://github.com/SwuduSusuwu) Your provided links are all private.

For now, https://github.com/SwuduSusuwu/SubStack/pull/46 shows `remark-lint`'s notices next to the commits which introduced the _Markdown_.
Will have to silence/dismiss those (to allow to merge), but will not for now, in the chance that those notices can help this issue to improve.

**Comment by JounQin** [comment]
> Warn when shortcut reference links are used.
> Note
>
> [no-shortcut-reference-link] Use the trailing [] on reference links

@SwuduSusuwu Why not just read the comments? This how `no-shortcut-reference-link` rule works?

**Comment by SwuduSusuwu** [comment]
> > Warn when shortcut reference links are used.
> > Note
> > [no-shortcut-reference-link] Use the trailing [] on reference links
> 
> [@SwuduSusuwu](https://github.com/SwuduSusuwu) Why not just read the comments? This how `no-shortcut-reference-link` rule works?

3 notices per footnote is excessive. In addition to the 2 above, `remark-lint` now also gives
```

[no-literal-urls] Don’t use literal URLs without angle brackets
Remark-lint (reported by Codacy)

Rule
Tool
Remark-lint (reported by Codacy)
Rule ID
remark-lint-no-literal-urls
Description

Warn for literal URLs in text.
URLs are treated as links in some Markdown vendors, but not in others.
To make sure they are always linked, wrap them in < (less than) and >
(greater than).
```
in response to footnotes (such as
```
[^3]: Asof [commmit b8850be10bdf5c18c1b0a4c98df2f86325c9c13e](https://github.com/SwuduSusuwu/SubStack/commit/b8850be10bdf5c18c1b0a4c98df2f86325c9c13e](./.ssh/sha256.sig) (and the [`./.ssh/allowed_signers`](./.ssh/allowed_signers) which `sha256.sig` produces) have a new certificate. This footnote was last altered by the first commit which uses that new certificate. **TODO**; warn if new commits use old certificates.
```
, which do not have literal URLs.)

> Please don't put your huge PR as reproduction next time, provide a minimal but runnable reproduction repo instead.

How? Do not know how to allow public access to individual notices (such as https://github.com/SwuduSusuwu/SubStack/security/code-scanning/2668).
Thought that those were public, but now know that those are not (for unknown reasons).

> You can config Codacy with `.remarkrc` config for this part.

How? _Codacy_ does not respond to issues about this.

**Comment by JounQin** [comment]
This is Codacy's issue, not `remark-lint`'s.

**Comment by SwuduSusuwu** [comment]
> This is Codacy's issue, not `remark-lint`'s.

_Codacy_ uses `remark-lint`'s defaults.
Those defaults should not give notices about literal URLs which do not occur.

**Comment by JounQin** [comment]
Codaycy should read `remarkrc` configs:

<img width="1598" alt="Image" src="https://github.com/user-attachments/assets/c866a3d6-e366-44e5-a645-df26e02c1b6d" />
